discovergy/discovergy.py

```diff
@@ -127,30 +138,46 @@ def login(self, email, password):
         :return: True on success, False on failure
         :rtype: bool """
 
+        self._fetch_consumer_tokens()
+        request_tokens = self._fetch_request_token()
+
         try:
-            self._fetch_consumer_tokens()
-            request_tokens = self._fetch_request_token()
             resource_owner_key = request_tokens["token"]
+        except Exception as e:
+            logger.error('Missing token in request_tokens: %s', e)
+            return False
```

Comment:
See? Here you are returning `False` if something went wrong. We probably need this in the above cases, too.